src/main/kotlin/no/nav/familie/ef/sak/service/JournalføringService.kt

```diff
@@ -40,19 +45,48 @@ class JournalføringService(private val journalpostClient: JournalpostClient,
     fun fullførJournalpost(journalføringRequest: JournalføringRequest, journalpostId: String): Long {
         val behandling: Behandling = hentBehandling(journalføringRequest)
         val journalpost = hentJournalpost(journalpostId)
+        val fagsak = fagsakService.hentFagsak(journalføringRequest.fagsakId)
 
-        settSøknadPåBehandling(journalpostId, behandling.fagsakId, behandling.id)
+        settSøknadPåBehandling(journalpostId, fagsak, behandling.id)
         knyttJournalpostTilBehandling(journalpost, behandling)
 
-        val eksternFagsakId = fagsakService.hentEksternId(journalføringRequest.fagsakId)
-        oppdaterJournalpost(journalpost, journalføringRequest.dokumentTitler, eksternFagsakId)
+        oppdaterJournalpost(journalpost, journalføringRequest.dokumentTitler, fagsak.eksternId.id)
         ferdigstillJournalføring(journalpostId, journalføringRequest.journalførendeEnhet)
         ferdigstillJournalføringsoppgave(journalføringRequest)
 
         return opprettSaksbehandlingsoppgave(behandling, journalføringRequest.navIdent)
 
     }
 
+    fun hentSøknadFraJournalpostForOvergangsstønad(journalpostId: String) : SøknadOvergangsstønad {
+       val dokumentinfo =
+        hentJournalpost(journalpostId).dokumenter
+                ?.first {
+                    DokumentBrevkode.OVERGANGSSTØNAD == DokumentBrevkode.fraBrevkode(it.brevkode.toString()) && harOriginalDokument(it)
+                } ?: error("Fant ingen søknad")
+
+        return journalpostClient.hentOvergangsstønadSøknad(journalpostId, dokumentinfo.dokumentInfoId)
+    }
+
+    fun hentSøknadFraJournalpostForBarnetilsyn(journalpostId: String) : SøknadBarnetilsyn {
+        val dokumentinfo =
+                hentJournalpost(journalpostId).dokumenter
+                        ?.first {
+                            DokumentBrevkode.BARNETILSYN == DokumentBrevkode.fraBrevkode(it.brevkode.toString()) && harOriginalDokument(it)
+                        } ?: error("Fant ingen søknad")
+
+        return journalpostClient.hentBarnetilsynSøknad(journalpostId, dokumentinfo.dokumentInfoId)
+    }
+
+    fun hentSøknadFraJournalpostForSkolepenger(journalpostId: String) : SøknadSkolepenger {
+        val dokumentinfo =
+                hentJournalpost(journalpostId).dokumenter
+                        ?.first {
+                            DokumentBrevkode.SKOLEPENGER == DokumentBrevkode.fraBrevkode(it.brevkode.toString()) && harOriginalDokument(it)
+                        } ?: error("Fant ingen søknad")
+
+        return journalpostClient.hentSkolepengerSøknad(journalpostId, dokumentinfo.dokumentInfoId)
+    }
```

Comment:
```suggestion
    fun hentSøknadFraJournalpostForOvergangsstønad(journalpostId: String): SøknadOvergangsstønad {
        val dokumentinfo = finnDokumentInfo(journalpostId, DokumentBrevkode.OVERGANGSSTØNAD)
        return journalpostClient.hentOvergangsstønadSøknad(journalpostId, dokumentinfo.dokumentInfoId)
    }

    fun hentSøknadFraJournalpostForBarnetilsyn(journalpostId: String): SøknadBarnetilsyn {
        val dokumentinfo = finnDokumentInfo(journalpostId, DokumentBrevkode.BARNETILSYN)
        return journalpostClient.hentBarnetilsynSøknad(journalpostId, dokumentinfo.dokumentInfoId)
    }

    fun hentSøknadFraJournalpostForSkolepenger(journalpostId: String): SøknadSkolepenger {
        val dokumentinfo = finnDokumentInfo(journalpostId, DokumentBrevkode.SKOLEPENGER)
        return journalpostClient.hentSkolepengerSøknad(journalpostId, dokumentinfo.dokumentInfoId)
    }

    private fun finnDokumentInfo(journalpostId: String,
                                 dokumentBrevkode: DokumentBrevkode): no.nav.familie.kontrakter.felles.journalpost.DokumentInfo {
        return hentJournalpost(journalpostId).dokumenter?.first {
            dokumentBrevkode == DokumentBrevkode.fraBrevkode(it.brevkode.toString()) && harOriginalDokument(it)
        } ?: error("Fant ingen søknad")
    }
```

Reply:
Helt klart bedre løsning. I stedet for finnDokumentInfo ble metoden kalt hentOriginaldokument